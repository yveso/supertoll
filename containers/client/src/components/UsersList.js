import React from "react";

function UsersList({ users }) {
  return (
    <ol>
      {users.map((user, index) => (
        <li key={index}>{user.username}</li>
      ))}
    </ol>
  );
}

export default UsersList;
