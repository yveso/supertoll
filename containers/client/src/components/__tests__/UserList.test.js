import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import UsersList from "../UsersList";

const users = [
  { active: true, email: "max@maxundmoritz.de", id: 0, username: "Max" },
  { active: true, email: "moritz@maxundmoritz.de", id: 0, username: "Moritz" },
];

test("UsersList renders", () => {
  const wrapper = shallow(<UsersList users={users} />);
  const element = wrapper.find("li");
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe("Max");
});

test("UsersList renders snapshot", () => {
  const tree = renderer.create(<UsersList users={users} />).toJSON();
  expect(tree).toMatchSnapshot();
});
