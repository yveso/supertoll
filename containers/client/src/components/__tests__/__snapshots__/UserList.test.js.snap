// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders snapshot 1`] = `
<ol>
  <li>
    Max
  </li>
  <li>
    Moritz
  </li>
</ol>
`;
